service/src/main/java/datawave/microservice/query/QueryManagementService.java

```diff
@@ -0,0 +1,2602 @@
+                                    .withMetricType(QueryMetricType.DISTRIBUTED)
+                                    .build());
+                    // @formatter:on
+                } catch (Exception e) {
+                    log.error("Error updating query metric", e);
+                }
+            }
+        } catch (CloneNotSupportedException e) {
+            log.warn("Could not determine whether the query logic supports metrics");
+        }
+    }
+    
+    /**
+     * Stops, and restarts the specified query.
+     * <p>
+     * Reset can be called on any query, whether it's running or not. <br>
+     * If the specified query is still running, it will be canceled. See {@link #cancel}. <br>
+     * Reset creates a new, identical query, with a new query id. <br>
+     * Reset queries will start running immediately. <br>
+     * Auditing is performed before the new query is started.
+     *
+     * @param queryId
+     *            the query id, not null
+     * @param currentUser
+     *            the user who called this method, not null
+     * @return a generic response containing the new query id
+     * @throws NotFoundQueryException
+     *             if the query cannot be found
+     * @throws UnauthorizedQueryException
+     *             if the user doesn't own the query
+     * @throws QueryException
+     *             if query lock acquisition fails
+     * @throws QueryException
+     *             if the cancel call is interrupted
+     * @throws BadRequestQueryException
+     *             if parameter validation fails
+     * @throws BadRequestQueryException
+     *             if query logic parameter validation fails
+     * @throws UnauthorizedQueryException
+     *             if the user doesn't have access to the requested query logic
+     * @throws BadRequestQueryException
+     *             if security marking validation fails
+     * @throws BadRequestQueryException
+     *             if auditing fails
+     * @throws QueryException
+     *             if query storage fails
+     * @throws QueryException
+     *             if there is an unknown error
+     */
+    public GenericResponse<String> reset(String queryId, DatawaveUserDetails currentUser) throws QueryException {
+        log.info("Request: reset from {} for {}", ProxiedEntityUtils.getShortName(currentUser.getPrimaryUser().getName()), queryId);
+        
+        try {
+            // make sure the query is valid, and the user can act on it
+            QueryStatus queryStatus = validateRequest(queryId, currentUser);
+            
+            // cancel the query if it is running
+            if (queryStatus.isRunning()) {
+                cancel(queryStatus.getQueryKey().getQueryId(), true);
+            }
+            
+            // create a new query which is an exact copy of the specified query
+            TaskKey taskKey = duplicate(queryStatus, new LinkedMultiValueMap<>(), currentUser);
```

Comment:
Will this cause the queryId that taskKey has to be the same as queryId, or is duplicate going to generate a new id? And does this change behavior of what duplicate does in DataWave now?

Reply:
there are some behavior differences that we need to talk about, yes.

Reply:
just to be clear, I would like to hold off on talking about those until the query microservices are vetted.  This round of PRs is mostly about getting the initial code in and getting the webservice verified as still being good.  Does that sound ok to you?

Reply:
If you trace that duplicate call, you'll see that a new query Id is created.

Reply:
Also, FWIW this is explained in the docs for the duplicate call.